### ISSUE TITLE
Upload then immediately delete?

### ISSUE THREAD

**Issue by Ashpork** [lavinir/hassio-onedrive-backup#37]
I'm getting situations where files are uploaded, then immediately deleted. The files are already pretty big (they appear to carry other backups which were performed, so each backup makes the backed up file bigger for the next), which perhaps needs to be a seperate issue I'll post.

Also, once it has deleted the files, it then goes to reupload them, even though it is outside of allowed times.

My logs are attached and config below. Do you need any other information?

[de91e161_hassio_onedrive_backup_2023-01-18T11-08-59.284Z.log](https://github.com/lavinir/hassio-onedrive-backup/files/10445231/de91e161_hassio_onedrive_backup_2023-01-18T11-08-59.284Z.log)

> recovery_mode: false
local_backup_num_to_keep: 10
onedrive_backup_num_to_keep: 5
backup_interval_days: 2
backup_name: hass_backup
hass_api_timeout_minutes: 30
notify_on_error: true
exclude_media_folder: true
exclude_ssl_folder: true
exclude_share_folder: true
exclude_local_addons_folder: false
sync_paths: []
sync_interval_hours: 4
backup_passwd: redacted
backup_allowed_hours: 0-4, 23

**Comment by lavinir** [comment]
1. Regarding uploads happening outside of allowed hours. You are right this is a bug. Backups won't be created but if it finds local backups that were not uploaded yet it will still do them. I will add a fix for this.
2. I think I understand what's happening here. Can you confirm you have 5 newer backups that exist both locally and in OneDrive and the ones being uploaded and then deleted are the 5 older backups you still have locally ?

**Comment by lavinir** [comment]
Yes, I meant while it's uploading those backups you should be going over the 5 limit on OneDrive, then it will reach 10 and delete the 5 older backups it just pushed..

**Comment by KosieKramer** [comment]
I am seeing the same issue. It continually tries to upload the 6th newest backup (I have set to have 5 in OneDrive, 10 locally), then immediately deletes it from OneDrive. 5 minutes later the process repeats:

```
01/20/2023 07:13:21 Info: Backup interval configured to every 24 hours
01/20/2023 07:13:21 Info: Backups / Syncs will only run during these hours: 0,1
01/20/2023 07:13:24 Info: Checking backups
01/20/2023 07:13:24 Info: Retrieving existing local backups...
01/20/2023 07:13:25 Info: Retrieving existing online backups...
01/20/2023 07:13:25 Info: Last local backup Date: 01/20/2023 00:21:22
01/20/2023 07:13:25 Info: Last online backup Date: 01/20/2023 00:21:22
01/20/2023 07:13:25 Info: Found 1 backups to upload.
01/20/2023 07:13:25 Info: Uploading hass_backup_2023-01-15-00-02 (01/15/2023 00:02:51)
01/20/2023 07:13:25 Info: Fetching Local Backup (Slug:c68d1016)
01/20/2023 07:13:27 Info: Backup (c68d1016) fetched successfully
01/20/2023 07:13:28 Info: Starting file upload. (Size:1376737280 bytes. Attempt: 1/3)
01/20/2023 07:14:12 Info: Uploaded 10%
01/20/2023 07:14:55 Info: Uploaded 20%
01/20/2023 07:15:38 Info: Uploaded 30%
01/20/2023 07:16:19 Info: Uploaded 40%
01/20/2023 07:17:02 Info: Uploaded 50%
01/20/2023 07:17:43 Info: Uploaded 60%
01/20/2023 07:18:25 Info: Uploaded 70%
01/20/2023 07:19:07 Info: Uploaded 80%
01/20/2023 07:19:49 Info: Uploaded 90%
01/20/2023 07:20:31 Info: Uploaded 100%
01/20/2023 07:20:33 Info: Upload completed successfully
01/20/2023 07:20:35 Info: Found 1 backups to delete from OneDrive.
01/20/2023 07:20:35 Info: Deleting file: hass_backup_2023-01-15-00-02.xxxxxx.tar
01/20/2023 07:20:36 Info: Backup Interval Completed.
01/20/2023 07:25:37 Info: Checking backups
01/20/2023 07:25:38 Info: Retrieving existing local backups...
01/20/2023 07:25:38 Info: Retrieving existing online backups...
01/20/2023 07:25:38 Info: Last local backup Date: 01/20/2023 00:21:22
01/20/2023 07:25:38 Info: Last online backup Date: 01/20/2023 00:21:22
01/20/2023 07:25:38 Info: Found 1 backups to upload.
01/20/2023 07:25:38 Info: Uploading hass_backup_2023-01-15-00-02 (01/15/2023 00:02:51)
01/20/2023 07:25:38 Info: Fetching Local Backup (Slug:c68d1016)
01/20/2023 07:25:40 Info: Backup (c68d1016) fetched successfully
01/20/2023 07:25:41 Info: Starting file upload. (Size:1376737280 bytes. Attempt: 1/3)
01/20/2023 07:26:24 Info: Uploaded 10%
01/20/2023 07:27:07 Info: Uploaded 20%
01/20/2023 07:27:48 Info: Uploaded 30%
01/20/2023 07:28:33 Info: Uploaded 40%
01/20/2023 07:29:13 Info: Uploaded 50%
01/20/2023 07:29:55 Info: Uploaded 60%
01/20/2023 07:30:39 Info: Uploaded 70%
01/20/2023 07:31:21 Info: Uploaded 80%
01/20/2023 07:32:04 Info: Uploaded 90%
01/20/2023 07:32:47 Info: Uploaded 100%
01/20/2023 07:32:49 Info: Upload completed successfully
01/20/2023 07:32:51 Info: Found 1 backups to delete from OneDrive.
01/20/2023 07:32:51 Info: Deleting file: hass_backup_2023-01-15-00-02.xxxxxx.tar
01/20/2023 07:32:52 Info: Backup Interval Completed.
```

Config:
```
recovery_mode: false
local_backup_num_to_keep: 10
onedrive_backup_num_to_keep: 5
backup_interval_days: 1
backup_name: hass_backup
hass_api_timeout_minutes: 60
notify_on_error: true
exclude_media_folder: true
exclude_ssl_folder: false
exclude_share_folder: false
exclude_local_addons_folder: false
sync_paths: []
backup_passwd: xxxxxx
backup_allowed_hours: 0-1
backup_instance_name: xxxxxx
```

**Comment by lavinir** [comment]
As mentioned above, I believe I pinpointed the issue. Fix will be included in the next release.
In the meantime, this shouldn't happen if you have the OneDrive backup number above or equals to the local backup number.

**Comment by lavinir** [comment]
Should be resolved now in version 1.41
